app/forms/candidate_interface/immigration_right_to_work_form.rb

```diff
@@ -15,9 +15,22 @@ def self.build_from_application(application_form)
     def save(application_form)
       return false unless valid?
 
-      application_form.update(
+      attrs = {
         immigration_right_to_work: immigration_right_to_work,
-      )
+      }
+      if right_to_work_or_study? == true
+        attrs.merge!(
+          immigration_route: nil,
+          immigration_route_details: nil,
+        )
+      else
+        attrs.merge!(
+          immigration_status: nil,
+          immigration_status_details: nil,
+          immigration_entry_date: nil,
+        )
+      end
```

Comment:
might be a bit easier to just do 

```
if right_to_work_or_study? == true
   application_form.update!(
       immigration_right_to_work: immigration_right_to_work,
       immigration_status: nil,
       immigration_status_details: nil,
   )
else 
  application_form.update!(
       immigration_right_to_work: immigration_right_to_work,
       immigration_status: nil,
       immigration_status_details: nil,
       immigration_entry_date: nil,
end 
```
wdyt?

Reply:
Yes much better